adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -210,6 +210,9 @@ private void performAuthorityValidation(final AuthenticationRequest authenticati
             // check if it contains authority url as the key, if key exists, it means that the authority url validation has happened
             // for the authority already.
             if (!UrlExtensions.isADFSAuthority(authorityUrl) && !AuthorityValidationMetadataCache.containsAuthorityHost(authorityUrl)) {
+                //Check if there is network connection
+                HttpWebRequest.throwIfNetworkNotAvailable(mContext);
```

Comment:
I would put this throw inside PerformInstanceDiscovery after line 213, i think the only reason this is here because the Context object is not available to run the check. I would add that to the constructor of Discovery.

Reply:
That has the benefit that if the cache is prepopulated and there is an AT in the cache, we will return that successfully. Let me know if i missed something.